app/src/main/java/com/udacity/sandwichclub/DetailActivity.java

```diff
@@ -69,24 +69,34 @@ private void closeOnError() {
 
     private void populateUI(Sandwich sandwich) {
         // Resource intensive findViewById lives in onCreate for one time population
-        originTv.append(" " + sandwich.getPlaceOfOrigin());
+        if (sandwich.getPlaceOfOrigin().isEmpty()) {
+            originTv.append(" N/A");
```

Comment:
if a similar check happens for each sandwich attribute, the N/A handling could be pulled out into a separate helper function

Reply:
would need to be a generic to handle both List and String `.isEmpty()` functions, considered out of scope and possible future optimization if were to release in production